### PR TITLE
Issue 3 nasa api setup

### PR DESCRIPTION
**Pull request by himo05**

### CODE REVIEW

File: API_Nasa_Call.py [hunk 1]
```diff
@@ -1 +1,89 @@
...
+        # Checks if the date_pic is within the beginning_date and today_date.
+        if date_pic <= today_date and date_pic >= beginning_date:
+            # # Returns a date_pic based on the current date and the beginning date.
+            return date_pic
+
+
+'''This function Returns the url for a given date.'''
+def get_url(date):
+    # API KEY.
+    API_KEY = '5c651pUEm3faSaZkgZ8vi2sKF7zntRQ2oXXXVhrc'
```
Comment (claraj) [inline API_Nasa_Call.py:48]: No API keys in the repository!

File: API_Nasa_Call.py [hunk 2]
```diff
@@ -1 +1,89 @@
...
+'''This function set a range of valid dates 
+    if the user enter the right dates it will return the date_pic if not it will ask them to enter the right dates'''
+    
+    # Get a picture for a specific date
+def get_date(): 
+
+    today_date = date.today() 
+    beginning_date = datetime.date(1995, 6, 16)  
+    
+    # recursively iterate over a list of False.
```
Comment (claraj) [inline API_Nasa_Call.py:20]: What does this comment mean?

File: API_Nasa_Call.py [hunk 3]
```diff
@@ -1 +1,89 @@
...
+'''This function Fetch the image URL from the given url.'''
+def fetch_image_url(url):
+    
+    try:
+        response = requests.get(url).json()
+        image_url = response['url']
+        return image_url
+    except:
+        print('Error extracting image')
+        return 'except'
```
Comment (claraj) [inline API_Nasa_Call.py:67]: For future reference - do something more useful in the exception handling

File: API_Nasa_Call.py [hunk 4]
```diff
@@ -1 +1,89 @@
...
+            return date_pic
+
+
+'''This function Returns the url for a given date.'''
+def get_url(date):
+    # API KEY.
+    API_KEY = '5c651pUEm3faSaZkgZ8vi2sKF7zntRQ2oXXXVhrc'
+    
+    # API URL for a planetary apod
+    url = f'https://api.nasa.gov/planetary/apod?api_key={API_KEY}&date={date}'
```
Comment (claraj) [inline API_Nasa_Call.py:51]: A dictionary of query params is cleaner

File: API_Nasa_Call.py [hunk 5]
```diff
@@ -1 +1,89 @@
...
+
+        
+
+'''This function Fetch the image URL from the given url.'''
+def fetch_image_url(url):
+    
+    try:
+        response = requests.get(url).json()
+        image_url = response['url']
+        return image_url
```
Comment (claraj) [inline API_Nasa_Call.py:64]: Do you need other data from the API, like the description text?

File: API_Nasa_Call.py [hunk 6]
```diff
@@ -1 +1,89 @@
-# Calls Nasa's API to return a picture of the APOD and return the link to the image for now until Flask is setup.
\ No newline at end of file
+# Calls Nasa's API to return a picture of the APOD and return the link to the image for now until Flask is setup.
+
+'''This API ask a user to enter a specific date between beginning_date which is 1995-6-16 until today_date the current day
```
Comment (claraj) [inline API_Nasa_Call.py:3]: Python docstrings typically use triple double quotes """

**Review by claraj — COMMENTED** [review]
Couple of notes just from reading your code, I haven't tried running your code.

This module should have one job to do - get the data from the API.  User input, validation, is obviously important but it should be somewhere else in the program.

File: API_Nasa_Call.py (hunk 2 above)
Comment (himo05) [reply API_Nasa_Call.py:20]: The loop will only stop when the condition is false I might explained it wrong.

File: API_Nasa_Call.py (hunk 6 above)
Comment (himo05) [reply API_Nasa_Call.py:3]: I'll fix that. Thanks for the feedback.

File: API_Nasa_Call.py (hunk 5 above)
Comment (himo05) [reply API_Nasa_Call.py:64]: I thought we only trying to get an image but sure I can add that too.

Comment (himo05) [reply API_Nasa_Call.py:64]: @claraj I got it now like, what is this picture? Thanks for the feedback

File: API_Nasa_Call.py (hunk 4 above)
Comment (himo05) [reply API_Nasa_Call.py:51]: I'll make the changes. Thanks the feedback.

**Review by Joeroble — APPROVED** [review]
Initial setup looks good, approved.

**Pull request merged**